### PR TITLE
[f] TAC-75 - Ingest daily report emails via special Missive webhook endpoint

### PR DESCRIPTION
**Pull request by kyphan1**
[TAC-75](https://linear.app/pdw/issue/TAC-75/ingest-daily-report-emails-via-special-missive-webhook-endpoint)

### CODE REVIEW

File: src/missive.js [hunk 1]
```diff
@@ -1,4 +1,7 @@
-const { openai } = require("./openai");
+const { JSDOM } = require("jsdom");
+const { supabaseTachio } = require("./supabaseclient");
+const DAILY_REPORT_REGEX = /Hi.*What has the team done since the last call\/email regarding this project\??(.*)What will the team do between now and the next call\/email regarding this project\??(.*)What impedes the team from performing their work as effectively as possible\??(.*)How much time have we spent today\??(.*)How much time have we spent this week\??(.*)How much time have we spent this month\??(.*)Our team today:?(.*)Regards/;
```
Comment (rajivsinclair) [inline src/missive.js]: @kyphan1 I have a couple suggestions:

- Can you please add a separate section for "design" items which begins with the following heading: "What has the design team done today?"

- We should extract the `[Design] # billable hour(s)` as a separate section after `How much time have we spent today/this week/this month?`
Screenshot 2024-04-12 at 4 42 39 PM

- Please do not extract the sections: 
 `How much time have we spent this week` 
 `How much time have we spent this month`
Screenshot 2024-04-12 at 4 45 02 PM

Comment (kyphan1) [reply src/missive.js]: `We should extract the [Design] # billable hour(s) as a separate section after How much time have we spent today/this week/this month?`

Based on this, are you suggesting to create a separate column for `design`, or should we use a single array column?

Comment (rajivsinclair) [reply src/missive.js]: I think we should create a separate column for `design`.

**Comment by kyphan1** [comment]
This PR needs to be merged together with https://github.com/PublicDataWorks/tachio-db/pull/2

File: src/missive.js (hunk 1 above)
Comment (kyphan1) [reply src/missive.js]: It's been done.

**Comment by rajivsinclair** [comment]
- [x] Can you please add a separate section for any `design` items within the `What has the team done …?` header?

**Comment by rajivsinclair** [comment]
- [x] Please do not extract the sections:
`How much time have we spent this week`
`How much time have we spent this month`
IMG_2276

**Comment by kyphan1** [comment]
> * [ ]  Can you please add a separate section for any `design` items within the `What has the team done …?` header?

It's done. I added another regex to detect it 
`const DESIGNER_REGEX = /\[Design].*?billable (?:hour|day)\(s\)/;`

**Comment by kyphan1** [comment]
> * [x]  Please do not extract the sections:
>       `How much time have we spent this week`
>       `How much time have we spent this month`
> 
> 
> IMG_2276

You can see from the code that I removed the parenthesis around these headers, so this data won't be captured.

**Comment by kyphan1** [comment]
> * [ ]  Can you please add a separate section for any `design` items within the `What has the team done …?` header?

Can you help trigger another daily report that includes this `design` item?

**Comment by rajivsinclair** [comment]
Here's an example:
example

Let me know if you think we should ask the team to adjust their template in any way.

**Comment by kyphan1** [comment]
@rajivsinclair  It's done. I also captured Design `willDo`

https://github.com/PublicDataWorks/tachio/assets/95199161/0e80638e-57be-4657-84a8-e5545d08d6d1

**Pull request merged**